pkg/knowledge_base2/properties/int_property.go

```diff
@@ -65,6 +65,11 @@ func (i *IntProperty) Parse(value any, ctx knowledgebase.DynamicContext, data kn
 	if val, ok := value.(int); ok {
 		return val, nil
 	}
+	if val, ok := value.(float32); ok {
+		return int(val), nil
+	} else if val, ok := value.(float64); ok {
+		return int(val), nil
+	}
```

Comment:
Not sure I'm a fan of `3.2` being a valid integer and truncating versus throwing a validation error. I don't mind `3.0` getting converted though.

Reply:
so you just want to validate that the float value ends in .0?

Reply:
Well, it's not quite that simple in floating point. Probably just check:
```
ival := int(val)
if abs(float64(ival) - val) > EPSILON { // EPSILON is some low number, like 0.00001
    return 0, fmt.Errorf("cannot convert non-integral float to int: %f", val)
}
```